corehq/apps/app_manager/views/forms.py

```diff
@@ -981,7 +981,7 @@ def _module_hierarchy(module):
 def get_form_datums(request, domain, app_id):
     form_id = request.GET.get('form_id')
     try:
-        datums = _get_form_datums(domain, app_id, form_id)
+        datums = _get_form_link_datums(domain, app_id, form_id)
```

Comment:
If `get_form_datums` also just returns "form_link_datums", does it make sense to rename that as well? This is coming from a place of not understanding the details of `_get_form_link_datums` and why we only return datums where `requires_selection` is true as someone reading through this code for the first time. Maybe it could be useful to document that somewhere?

Reply:
yeah I think it makes sense to rename the view (and URL and what not) too.  Or at least keep them consistent.  I'll do that.

Though I think I might actually walk back the changes to this function since it doesn't apply to advanced mode datums and modules aren't linkable aside from in that mode that.  This work has seen a decent bit of churn as we try to get something working (it's still being tested on staging right now, actually, though I think the current issues are on the formplayer side of things).  An earlier version intended to extend the manual linking to modules, but it just didn't work in practice (we know that the existing version of this feature has edge cases that fail).

> why we only return datums where requires_selection is true

This is documented a bit here:
https://github.com/dimagi/commcare-hq/blob/f51b264bfbe167b0440e67da464639bfa7e54131/corehq/apps/app_manager/models.py#L780-L794

And in a broader context on [readthedocs](https://commcare-hq.readthedocs.io/apps/advanced_app_features.html#end-of-form-navigation-and-form-linking).

The gist of it is that to navigate to a form or menu, you have to provide datums that "require selection", like choosing a case from a case list.  Those that don't require selection don't need to be provided manually, they can be inferred.